app/scheme.go

```diff
@@ -117,3 +118,15 @@ func (a *App) IsPhase2MigrationCompleted() *model.AppError {
 
 	return nil
 }
+
+func (a *App) SchemeGenerator(batchSize int) func() []*model.Scheme {
```

Comment:
I don't like the name of the function, looks confusing, maybe `SchemePaginator`? or `PaginatedSchemes`?

Reply:
Yes, I prefer SchemesIterator, because SchemeGenerator sounds like is generating new schemes in any way, but the Iterator have only one possible meaning :)